lang/java/reef-applications/reef-vortex/src/main/java/org/apache/reef/vortex/driver/VortexDriver.java

```diff
@@ -76,7 +79,9 @@ private VortexDriver(final EvaluatorRequestor evaluatorRequestor,
                        @Parameter(VortexMasterConf.WorkerMem.class) final int workerMem,
                        @Parameter(VortexMasterConf.WorkerNum.class) final int workerNum,
                        @Parameter(VortexMasterConf.WorkerCores.class) final int workerCores,
-                       @Parameter(VortexMasterConf.NumberOfVortexStartThreads.class) final int numOfStartThreads) {
+                       @Parameter(VortexMasterConf.NumberOfVortexStartThreads.class) final int numOfStartThreads,
+                       @Parameter(VortexMasterConf.InputCodec.class) final Codec inputCodec,
```

Comment:
Another approach could be keeping serialized output until we actually need: the result is passed to the `VortexFuture` in `VortexWorkerManager.taskletCompleted(taskletId, result)`. As `Tasklet` maintains `VortexFunction` inside it, we can directly decode the output here. This way, the message receiver does not need to know about the input/output codec.
@bgchun How does this sound?